src/libtools/application.cc

```diff
@@ -59,25 +59,38 @@ void Application::ShowHelpText(std::ostream &out) {
 }
 
 void Application::ShowManPage(std::ostream &out) {
+  out << FormatManPage<globals::man>();
+}
+
+void Application::WriteManFile() {
+  ofstream man_file;
+  man_file.open(ProgramName() + "_man.rst");
+  man_file << FormatManPage<globals::manrst>();
+  man_file.close();
+}
 
-  out << boost::format(globals::man::header) % ProgramName() % VersionString();
-  out << boost::format(globals::man::name) % ProgramName() % globals::url;
-  out << boost::format(globals::man::synopsis) % ProgramName();
+template <typename T>
```

Comment:
What is the idea behind the template parameter?

Reply:
Printing the manual in `groff`  or writting in `rst` format uses the same functionality but different "data structures". There are now two structures with the data for the manual pages: one in `groff` format and the other in `rst`. They are called `globals::man` and `globals::manrst`, respectively.

Reply:
okay, that makes sense.

Reply:
but normally `T` is used for generic template parameters, I think in this case we can give a more explicit name to it.

Reply:
but I am not bent on it. So only do it if you agree

Reply:
I agree that we can give a more generic type name because is either `man` or `manrst` and both are documentation format. So, what about:
```cc
template <typename Format>
```

Reply:
nice